pkg/repository/udmrepo/kopialib/backend/azure.go

```diff
@@ -18,43 +18,37 @@ package backend
 
 import (
 	"context"
+	"encoding/base64"
 
 	"github.com/kopia/kopia/repo/blob"
-	"github.com/kopia/kopia/repo/blob/azure"
+	"github.com/pkg/errors"
 
 	"github.com/vmware-tanzu/velero/pkg/repository/udmrepo"
+	"github.com/vmware-tanzu/velero/pkg/repository/udmrepo/kopialib/backend/azure"
 )
 
 type AzureBackend struct {
-	options azure.Options
+	option azure.Option
 }
 
 func (c *AzureBackend) Setup(ctx context.Context, flags map[string]string) error {
-	var err error
-	c.options.Container, err = mustHaveString(udmrepo.StoreOptionOssBucket, flags)
-	if err != nil {
-		return err
+	// As pkg/util/azure.NewStorageClient(config) is used in both repository and plugin,
+	// the caCert isn't encoded when passing to the plugin, so we need to decode the caCert
+	// before passing the config into the NewStorageClient()
+	if flags[udmrepo.StoreOptionCACert] != "" {
```

Comment:
I think we need to either keep the caCert encoded or use []byte to take it. As we discussed, some formats of the caCert could not be taken by string, there may be functionality lose if we decode it and assign it to a string.

I understand the conflict between the repo code and the plugin code, and I think the problem is on the plugin code --- it didn't consider all the caCert formats. So let's whether see we can fix this problem on plugin side.

Reply:
As the comment said, the `AzureBackend.Connect()` calls `pkg/util/azure.NewStorageClient(config map[string]string)`.
The `pkg/util/azure.NewStorageClient(config map[string]string)` is also called by the plugin code while the plugin framework passes a string caCert to the plugin code, so we cannot use `[]byte` caCert.

I'll open an issue to track the caCert issue.

Reply:
If we cannot fix the problem on the plugin side, I think we need to isolate the calls so that the repo's functionality won't break. Possibly something like this:

- Let the plugin call `pkg/util/azure.NewStorageClientA(config map[string]string)`
- Add another function NewStorageClientB which accepts []byte for caCert
- NewStorageClientA calls NewStorageClientB and converts the string caCert to []byte
- The repo flow calls NewStorageClientB directly